components/NacelleButton.vue

```diff
@@ -0,0 +1,135 @@
+<template>
+  <div :class="block ? 'block' : null" class="nacelle-button">
+    <button :class="color" @click="clickHandler()">
+      <span>
+        <slot />
+      </span>
+      <template v-if="!noIcon">
+        <div v-if="icon" v-html="icon" class="nacelle-button-icon"></div>
+        <img v-else src="/nacelle-arrow-right.svg" alt="arrow-right" />
+      </template>
+    </button>
```

Comment:
Should we extend this component to also render a nuxt-link styled as a button for accessibility? We could add a prop for `url` and use that to determine when to be a link vs a button.

```
<nuxt-link v-if="url">
  <slot />
</nuxt-link>

<button v-else>
  <slot />
</button>
```

Reply:
That would also allow us to simplify the `clickHandler()` function

Reply:
I agree with you! This is also just a copy from the Nacelle website component. I thought it was weird too because it seems like this field in the Contentful block would almost always be to a url as well. @levimichael I can certainly replace the button so that it resolves to a regular link tag, do you have an opinion?

Reply:
I don't see Nuxt as a part of this codebase. Would it be worth adding that just for the `nuxt-link`? I do agree that some accessibility treatment would be great.

Reply:
@Yahsflame You're right. This is vuepress, so it would get the `router-link`

Reply:
@darzouras I think today it probably makes the most sense to resolve a regular link tag that is styled-- especially since it has to navigate to a different whole website technically.

Reply:
Update: replaced button with an a tag (as router-link is meant for internal links only!), and styled to be identical otherwise! @drewclem @Yahsflame @levimichael

Reply:
Nice work! And good catch on the router-link!